File: JsCore/JavaScript Advanced/Classes and Class Members/Circle.js
class Circle {
    constructor (radius) {
        this.radius = radius;
    }
    get diameter() {
        return 2 * this.radius;
    }
    set diameter(diameter) {
        this.radius = diameter / 2;
    }

    get area() {
        return Math.PI * this.radius * this.radius;
    }
}
let c = new Circle(2);
console.log(c.radius);
console.log(c.diameter);
console.log(c.area);
c.diameter = 6;
console.log(c.radius);

File: C# Fundamentals/C# OOP Basics/Polymorphism Lab/3. Shapes/Models/Circle.cs
using System;

public class Circle : Shape
{
    private double _radius;

    public Circle(double radius)
    {
        this.Radius = radius;
    }

    private double Radius
    {
        get => _radius;
        set => _radius = value;
    }


    public override double CalculatePerimeter()
    {
        return 2 * Math.PI * this.Radius;
    }

    public override double CalculateArea()
    {
        return Math.PI * Math.Pow(this.Radius, 2);
    }

    public override string Draw()
    {
        return base.Draw() + "Circle";
    }
}
